build/patches/Ask-user-before-closing-all-tabs.patch

```diff
@@ -18,12 +18,12 @@ diff --git a/chrome/android/java/src/org/chromium/chrome/browser/ChromeTabbedAct
  
  import org.chromium.base.CallbackController;
  import org.chromium.base.CommandLine;
-@@ -2054,8 +2055,23 @@ public class ChromeTabbedActivity extends ChromeActivity<ChromeActivityComponent
-             ApplicationLifetime.terminate(false);
-         } else if (id == R.id.close_all_tabs_menu_id) {
-             // Close both incognito and normal tabs
--            getTabModelSelector().closeAllTabs();
--            RecordUserAction.record("MobileMenuCloseAllTabs");
+@@ -2065,8 +2066,23 @@ public class ChromeTabbedActivity extends ChromeActivity<ChromeActivityComponent
+                 CloseAllTabsDialog.show(this, getModalDialogManagerSupplier(),
+                         () -> getTabModelSelector().closeAllTabs(), /*isIncognito=*/false);
+             } else {
+-                // Close both incognito and normal tabs.
+-                getTabModelSelector().closeAllTabs();
```

Comment:
upstream added new flag `kCloseAllTabsModalDialog` disabled by default.
this patch can be removed and the flag enabled